### ISSUE TITLE
open image from a big folder is overly slow

### ISSUE THREAD

**Issue by roger21** [easymodo/qimgv#156]
hey,

so if i want to open an image with qimgv from a folder that has a lots of images (about 1-2 thousands) it takes a while, making qimgv unusable in that context

i tried disabling caching and prepossessing in the settings (not being sure what it does) it changes nothing

**Comment by easymodo** [comment]
Weird, it starts quite fast for me. What OS?  SSD/HDD? And also you should update if it's older than v0.8.6

>i tried disabling caching and prepossessing

Keep that enabled, shouldn't influence load times.

**Comment by roger21** [comment]
well ok indeed it was not a common case but since no other viewer (i am currently testing a bunch) did that i assumed the file location didn't matter

it was other the network (smb share) and indeed copied to local (still about 2 thousand) it's instantaneous

anyway, there is still an issue here but limited (again, others viewers don't do that)

**Comment by EdwinKM** [comment]
What i noticed, and i tested a lot of viewers. When i just double click, most of them are creating a complete listing of the directory where the picture resides. And only after that they actually show the image. This can be really annoying on network shares. Yes, please buffer/preload/index and create a playlist of all images (in a separate thread). But first of all. Show the selected picture.

**Comment by easymodo** [comment]
>Yes, please buffer/preload/index and create a playlist of all images (in a separate thread). But first of all. Show the selected picture.

Just fixed a bug which caused increased load time (by 50-100ms) on very large directories. 
A bunch of optional ui was loaded before the image.
I'll also try to do threaded directory listing but it may break my spaghetti code

**Comment by EdwinKM** [comment]
well, i understand why most viewers do it. Without it you have some difficult extra challenges:
The user can supply a random item (which normally could be in the middle of a sorted playlist). You have to open it. You have to locate it in the "playlist" (index). What to do with sorting while finding all new files? On the fly updating the playlist GUI. The user can already press "next" before all other items are found and added. The only sensible way is to disable "sorting" in this case?

**Comment by easymodo** [comment]
maybe just block (or buffer) user input, then once directory is loaded - execute it
the delay shouldnt be noticeable (it's like half a second on HUGE directories >10k files)

**Comment by roger21** [comment]
there have been a huge improvement with the last release 0.9.1, making it bearable in my case

**Comment by nasbdh9** [comment]
Using version 9.1, it takes about 5 seconds to open a file in a folder of about 1,000 pictures. When the file increases to about 5,000 to 10,000, it takes about 20 seconds to open.

**Comment by easymodo** [comment]
works fine over here.. slow hdd, ~3k files, launches in about half a second
almost instant on ssd

i'll check where it might choke up, but again, its perfectly fine on my pc

**Comment by nasbdh9** [comment]
> works fine over here.. slow hdd, ~3k files, launches in about half a second
> almost instant on ssd
> 
> i'll check where it might choke up, but again, its perfectly fine on my pc

I read a file from a samba server with an all-solid-state drive. When using ImageGlass, the time required to open the file is normal (about 1 second)

**Comment by easymodo** [comment]
network shares are whole another story, yes it is currently does not work correctly
wasnt my use case so i kinda postponed working on it
